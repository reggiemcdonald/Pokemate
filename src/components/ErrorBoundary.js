import React from "react";
import {
    Text,
    View,
} from "react-native";
import Ionicons from "react-native-vector-icons/Ionicons";
import styles from "../library/styles";

/**
 * **************************
 * A Error FallBack Component
 * **************************
 * Renders when an error is caught
 *
 */
export default class ErrorBoundary extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            hasError: false,
            displayIcon: this.props.displayIcon ? this.props.displayIcon : <Ionicons name={"ios-alert"}/>,
            message: this.props.message ? this.props.message : "Oops! Something happened :("
        }
    }

    static getDerivedStateFromError(error) {
        return {
            hasError: true,
            error: error
        }
    }

    /**
     * Log the error that was caught
     * @override
     * @param error
     * @param errorInfo
     */
    componentDidCatch(error: Error, errorInfo: React.ErrorInfo): void {
        this.setState({
            hasError: true,
            error: error,
            errorInfo: errorInfo
        });
    }

    render() {
        if (this.state.hasError) {
            return(
                <View style={styles.containerCentered}>
                    {this.state.displayIcon}
                    <Text>{this.state.message}</Text>
                </View>
            )
        } else {
            return this.props.children;
        }
    }


}