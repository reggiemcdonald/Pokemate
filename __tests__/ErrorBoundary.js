import React from "react";
import renderer from "react-test-renderer";
import ErrorBoundary from "../src/components/ErrorBoundary";
import {View} from "react-native";

describe("Error Boundary: Should Render", () => {
    it("Should render without crashing", () => {
        const tree = renderer.create(
            <ErrorBoundary>
                <View></View>
            </ErrorBoundary>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});