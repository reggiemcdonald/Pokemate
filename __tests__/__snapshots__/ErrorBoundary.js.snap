// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary: Should Render Should render without crashing 1`] = `<View />`;
